dbt/adapters/redshift/connections.py

```diff
@@ -90,30 +94,44 @@ def fresh_transaction(self, name=None):
 
     @classmethod
     def fetch_cluster_credentials(
```

Comment:
Your changes to this function make sense, but I want to flag it for future refactoring if you could throw this comment on it.

```suggestion
    # TODO with the addition of the `serverless` field there are non non-nonsensical ways to call this function. It deserves a refactor.
    @classmethod
    def fetch_cluster_credentials(
```